test/annexB/templates/legacy-octal-escape-sequence-non-strict.js

```diff
@@ -0,0 +1,13 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 12.2.8
+description: >
+    The expression within the template should be evaluated according to the
+    semantics of the surrounding context.
+    The SV of EscapeSequence :: HexEscapeSequence is the SV of the
+    HexEscapeSequence.
+flags: [noStrict]
+---*/
+
+assert.sameValue(`${'\07'}`, '');
```

Comment:
I'm still kind of keen on using hex escapes or unicode escapes for the expected values here --- I know it means using another language feature to test this one, but it's a lot easier to see what's going on on github, and in many text editors, too.